### PR TITLE
Pset_UndergroundExcavationCommon

### PR DESCRIPTION
**Pull request by larswik**
# Pset: Pset_UndergroundExcavationCommon

## Description: 

Common properties for underground excavation elements

| Property                    | Datatype                          | Description                                                  |
| --------------------------- | --------------------------------- | ------------------------------------------------------------ |
| UndergroundExcavationMethod | PEnum_UndergroundExcavationMethod | Method of excavation                                         |
| ExcavationMethodOther       | IfcLabel                          | A user defined underground excavation method when UndergroundExcavationMethod = USERDEFINED |
| SupportDescription          | IfcLabel                          | Description of Support                                       |
| SupportClass       | IfcLabel                          | Specifies a class defining the support measures for underground excavation.  |
| ExcavationClass    | IfcLabel                          | Specifies a class defining the support measures for underground excavation.  |
| RoundLength                 | IfcPositiveLengthMeasure          | Length of individual round or drive                          |

## Enumeration: PEnum_UndergroundExcavationMethod

| Constant                     | Description |
| ---------------------------- | ----------- |
| HAND                         | Manual excavation method using hand tools.  |
| EXCAVATOR                    | Mechanized method utilizing excavator machinery for efficient large-scale underground earthmoving.  |
| JACKHAMMER                   | Pneumatic or hydraulic percussion tool breaking rock or concrete.  |
| RIPPER                       | Equipment with a toothed tool to break and loosen hard soil.  |
| DRILLANDBLAST                | A method of desintegrating rock by drilling small diameter holes on a planned layout, packing these with explosives and then firing to a fixed program to shatter the rock in a desired form.  |
| ROADHEADER                   | Machine with rotating cutter head specifically designed for the excavation of roadways, tunnels, and underground chambers.  |
| TBMOPENWITHGRIPPERS          | An open rock tunnel boring machine which generally utilizes roller disc cutters as excavation tools and which moves forward by reacting  against the tunnel walls through a hydraulic gripper reaction system. |
| TBMCLOSEDWITHGRIPPERS          | A closed-face shield which generally utilizes roller disc cutters as excavation tools and which moves forward by reacting  against the tunnel walls through a hydraulic gripper reaction system. |
| TBMEARTHPRESSUREFACESUPPORT  | A closed-face shield designed for tunnelling in fine-grained soils by trapping excavated materials against the face and removing them at a rate slow enough to maintain pressures that counterbalance earth pressures exerted against the cutterhead and prevent ingress of water; which moves forward by reacting  against an installed lining. |
| TBMSLURRYPRESSUREFACESUPPORT | A closed-face shield designed for tunnelling in very soft, wet, or running ground by use of circulating, pressurized clay slurry against the face to counterbalance earth pressures exerted against the cutterhead, prevent ingress of water and also to carry away the excavated material; which moves forward by reacting  against an installed lining. |
| TBMAIRPRESSUREFACESUPPORT    | A closed-face shield designed for tunnelling in unstable soils with the presence of water where compressed air is used to counterbalance earth pressures exerted against the cutterhead; which moves forward by reacting against an installed lining. |
| TBMMIXSHIELD                 | A closed-face shield  with a mixed face support system that combines various methods to counterbalance earth pressures exerted against the cutterhead; which moves forward by reacting against an installed lining. |
| SHAFTBORINGMACHINE           | Specialized machine for boring vertical or inclined shafts.  |
| RAISEDRILL                   | Type of shaft boring machine that enlarges a pilot hole by reaming to the full shaft diameter from the bottom to the top of the shaft. |
| USERDEFINED                        |             |

### CODE REVIEW

File: IFC4x3/Properties/e/ExcavationClassification_0M$dI8P5T3I8$lTN_LlVyH/DocProperty.xml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (SergejMuhic) [inline IFC4x3/Properties/e/ExcavationClassification_0M$dI8P5T3I8$lTN_LlVyH/DocProperty.xml]: Should this not be modelled with [Classification Association](https://bsi-infraroom.github.io/IFC-Documentation-Tunnel/4_4_0_0/general/HTML/link/classification-association.htm)?

File: IFC4x3/Properties/r/RoundLength_2Px7GAEKr9oPOVi4hIhBDX/Documentation.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Length of individual round or drive.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/r/RoundLength_2Px7GAEKr9oPOVi4hIhBDX/Documentation.md:1]: What is a "Round"?

File: IFC4x3/Properties/s/SupportClassification_09LzP6dcn6OAdINBkUxEqS/DocProperty.xml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (SergejMuhic) [inline IFC4x3/Properties/s/SupportClassification_09LzP6dcn6OAdINBkUxEqS/DocProperty.xml]: Also, Classification Association?

**Review by SergejMuhic — COMMENTED** [review]
Constants and enums do not have documentation.

File: IFC4x3/Properties/e/ExcavationClassification_0M$dI8P5T3I8$lTN_LlVyH/DocProperty.xml (hunk 1 above)
Comment (larswik) [reply IFC4x3/Properties/e/ExcavationClassification_0M$dI8P5T3I8$lTN_LlVyH/DocProperty.xml]: I agree, but involving @JontyDunn in this conversation.

File: IFC4x3/Properties/r/RoundLength_2Px7GAEKr9oPOVi4hIhBDX/Documentation.md (hunk 2 above)
Comment (larswik) [reply IFC4x3/Properties/r/RoundLength_2Px7GAEKr9oPOVi4hIhBDX/Documentation.md:1]: Ping @JontyDunn.

File: IFC4x3/Properties/s/SupportClassification_09LzP6dcn6OAdINBkUxEqS/DocProperty.xml (hunk 3 above)
Comment (larswik) [reply IFC4x3/Properties/s/SupportClassification_09LzP6dcn6OAdINBkUxEqS/DocProperty.xml]: I agree, but involving @JontyDunn in this conversation.

File: IFC4x3/Properties/e/ExcavationClassification_0M$dI8P5T3I8$lTN_LlVyH/DocProperty.xml (hunk 1 above)
Comment (larswik) [reply IFC4x3/Properties/e/ExcavationClassification_0M$dI8P5T3I8$lTN_LlVyH/DocProperty.xml]: Rename to ExcavationClass. @JontyDunn or @hessf-ilf might improve the description, e.g. with an example. Right now it is "Specifies an excavation class used in underground excavation."

File: IFC4x3/Properties/r/RoundLength_2Px7GAEKr9oPOVi4hIhBDX/Documentation.md (hunk 2 above)
Comment (larswik) [reply IFC4x3/Properties/r/RoundLength_2Px7GAEKr9oPOVi4hIhBDX/Documentation.md:1]: Ping @hessf-ilf

File: IFC4x3/Properties/s/SupportClassification_09LzP6dcn6OAdINBkUxEqS/DocProperty.xml (hunk 3 above)
Comment (larswik) [reply IFC4x3/Properties/s/SupportClassification_09LzP6dcn6OAdINBkUxEqS/DocProperty.xml]: Rename to SupportClass. @JontyDunn or @hessf-ilf might improve the description, e.g. with an example. Right now it is "Specifies a class defining the support measures for underground excavation."

**Comment by larswik** [comment]
In PEnum_UndergroundExcavationMethod, I removed (and deleted) OTHER and added (re-used) USERDEFINED instead.

**Comment by larswik** [comment]
> Constants and enums do not have documentation.

Fixed in [d48ad31](https://github.com/bSI-InfraRoom/IFC-Specification/pull/570/commits/d48ad310dc37278bb55272f1a387eb3c23823d09)

File: IFC4x3/Properties/e/ExcavationMethodOther_1fa7O2cGn6_fZVIehnm$Jw/Documentation.md [hunk 4]
```diff
@@ -0,0 +1 @@
+A user defined underground excavation method when UndergroundExcavationMethod = USERDEFINED.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/e/ExcavationMethodOther_1fa7O2cGn6_fZVIehnm$Jw/Documentation.md]: Property name says Other.

File: IFC4x3/PropertyEnumerations/u/PEnum_UndergroundExcavationMethod.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (SergejMuhic) [inline IFC4x3/PropertyEnumerations/u/PEnum_UndergroundExcavationMethod.xml:1]: Does not have a description.

File: IFC4x3/PropertyConstants/c/COREDRILLING_3kAMQZBc19h9RiBVqacIVy.xml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="COREDRILLING_3kAMQZBc19h9RiBVqacIVy" Name="COREDRILLING" UniqueId="ee2966a3-2e60-49ac-96ec-2dfd249927fc">
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/c/COREDRILLING_3kAMQZBc19h9RiBVqacIVy.xml]: Apparently removed.

File: IFC4x3/PropertyConstants/d/DESTRUCTIVEDRILLING_0kW0UHgtH7WPGblh83KWIC.xml [hunk 7]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="DESTRUCTIVEDRILLING_0kW0UHgtH7WPGblh83KWIC" Name="DESTRUCTIVEDRILLING" UniqueId="2e800791-ab74-4781-9425-beb20352048c">
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/d/DESTRUCTIVEDRILLING_0kW0UHgtH7WPGblh83KWIC.xml]: Seems removed also.

File: IFC4x3/PropertyConstants/t/TBMCLOSEDWITHGRIPPERS_24tst5kj171RquDTEFq43J.xml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="TBMCLOSEDWITHGRIPPERS_24tst5kj171RquDTEFq43J" Name="TBMCLOSEDWITHGRIPPERS" UniqueId="84df6dc5-bad0-4705-bd38-35d38fd040d3">
+	<Documentation>A closed-face shield which generally utilizes roller disc cutters as excavation tools and which moves forward by reacting  against the tunnel walls through a hydraulic gripper reaction system.</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/t/TBMCLOSEDWITHGRIPPERS_24tst5kj171RquDTEFq43J.xml]:
```suggestion
	<Documentation>A closed-face shield which generally utilizes roller disc cutters as excavation tools and which moves forward by reacting against the tunnel walls through a hydraulic gripper reaction system.</Documentation>
```

File: IFC4x3/Properties/e/ExcavationMethodOther_1fa7O2cGn6_fZVIehnm$Jw/Documentation.md (hunk 4 above)
Comment (larswik) [reply IFC4x3/Properties/e/ExcavationMethodOther_1fa7O2cGn6_fZVIehnm$Jw/Documentation.md]: > Property name says Other.

I have no good proposal on what to call this and make it short?

Comment (larswik) [reply IFC4x3/Properties/e/ExcavationMethodOther_1fa7O2cGn6_fZVIehnm$Jw/Documentation.md]: We decided to remove this.

**Comment by larswik** [comment]
Should be ok now. Removed the ExcavationMethodOther property.

**Pull request merged**